ArchiSteamFarm/Core/ASF.cs

```diff
@@ -285,6 +285,31 @@ public static class ASF {
 				return null;
 			}
 
+			ArchiLogger.LogGenericInfo(Strings.VerifyingChecksumWithRemoteServer);
+
+			string? remoteChecksum = await ArchiNet.FetchBuildChecksum(newVersion, SharedInfo.BuildInfo.Variant).ConfigureAwait(false);
+
+			switch (remoteChecksum) {
+				case null:
+					// Timeout or error, refuse to update as a security measure
+					return null;
+				case "":
+					// Unknown checksum, release too new or actual malicious build published, no need to scare the user as it's 99.99% the first
+					ArchiLogger.LogGenericWarning(Strings.ChecksumMissing);
+
+					return SharedInfo.Version;
+			}
+
+			byte[] responseBytes = response.Content as byte[] ?? response.Content.ToArray();
+
+			string checksum = BitConverter.ToString(SHA512.HashData(responseBytes)).Replace("-", "", StringComparison.Ordinal);
```

Comment:
Compare them as raw bytes instead.

Reply:
It'd involve either converting string to bytes or bytes to string, either is work needed, but the upside of having it like that is the fact that every array of bytes converts to hex, and in case of string to bytes I'd need to handle a situation where it's not a hex to begin with, seems silly when we can just let the checksum fail instead by string comparison :shrug:

Reply:
Your server could reply with bytes if you wanted to.

Reply:
It could, but I'd still need to encode them somehow, and it seems like too much of a burden to consider for 128-chars hexadecimal string - it's not like it's some hot path that will execute more often than once per update anyway, which usually happens monthly.

Reply:
Plus it makes it easier to store in database and parse from `SHA512SUMS` attached with the release, I agree that both methods can be used and byte one could be more effective, but it's really not fundamentally wrong as it is right now imho, and I see some disadvantages of making my life harder.

Reply:
(Not to mention comparing 2 strings against equality is not that much slower than byte representation, especially when `Ordinal` culture is used)